enhancements/update/cluster-version-operator-x509-upstream-trust.md

```diff
@@ -0,0 +1,202 @@
+The CVO also retrieves release image signatures from [configured stores][update-keys] (and [other places](../oc/mirroring-release-image-signatures.md)) to establish trust for the recommended update targets.
+Both of those may require HTTPS calls to external services.
+
+In some cases, those HTTPS calls will use a [configured proxy](../proxy/global-cluster-egress-proxy.md).
+And currently there is no mechanism for configuring additional HTTP(S) signature stores.
+But the update recommendation service is configurable [with ClusterVersion's `upstream` property][api-upstream], and adminstrators may point that at internal services whose X.509 certificate is not signed by a certificate authority (CA) that the CVO trusts by default.
+
+## Motivation
+
+### Goals
+
+Administrators will be able to configure the CVO to communicate with their configured [`upstream`][api-upstream], even if the TLS terminates with a service that:
+
+* Uses an X.509 certificate signed by a CA that is not included in the CVO's default trust store.
+* Uses TLS connection properties (version, ciphers, etc.) that are not accepted by the CVO's default TLS profile.
+
+### Non-Goals
+
+This enhancement does not address components outside the CVO.
+[Enhancement 115][enhancement-115] proposed a general framework for configuring X.509 trust, but was [rejected][enhancement-115-rejection] in favor of per-component solutions.
+
+This enhancement does not address signature retrieval.
+Currently [the configured stores][update-keys] are hard-coded, so the CVO will only talk to an alternative TLS terminator when it is being [proxied](../proxy/global-cluster-egress-proxy.md).
+The fact that the current proxy configuration does not clearly address proxies defined at the network-routing level, where `httpsProxy` may be empty but additional trust still needs to be injected is out of scope.
+
+## Proposal
+
+A new config type `` will be added consuming the existing [`TLSSecurityProfile`][api-tls-security-profile] and [`ConfigMapNameReference`][api-config-map-name-reference].
+The `securityProfile` property and its comment are copied from [the existing `APIServerSpec` consumer][api-server-tls-security-profile] with the leading `TLS` dropped because `TLSConfig` is already TLS-specific.
+The `TrustedCA` property and its comment are based on [the existing `Proxy` consumer][api-proxy-trusted-ca].
+
+```go
+type TLSConfig {
+  // securityProfile specifies settings for TLS connections for externally exposed servers.
+  //
+  // If unset, a default (which may change between releases) is chosen. Note that only Old and
+  // Intermediate profiles are currently supported, and the maximum available MinTLSVersions
+  // is VersionTLS12.
+  // +optional
+  SecurityProfile *TLSSecurityProfile `json:"securityProfile,omitempty"`
+
+	// trustedCA is a reference to a ConfigMap containing a PEM-encoded X.509
```

Comment:
Looks like you have a good-ole tabs (below) vs. spaces (above) mismatch here.

Reply:
> Looks like you have a good-ole tabs (below) vs. spaces (above) mismatch here.

:facepalm: Fixed with 001a116 -> 47f96bb.